hayeon/programmers-0/OX퀴즈.js

```diff
@@ -0,0 +1,17 @@
+function solution(quiz) {
+  const result = [];
+
+  quiz.map((v) => {
+    const tmp = v.split(" ");
+
+    if (tmp[1] === "-") result.push(+tmp[0] - +tmp[2] === +tmp[4] ? "O" : "X");
+    else result.push(+tmp[0] + +tmp[2] === +tmp[4] ? "O" : "X");
+  });
+  return result;
+}
```

Comment:
```suggestion
function solution(quiz) {
  return quiz.map((v) => {
    const formula = v.split(" ").map((el) => (isNaN(+el) ? el : +el));

    if (formula[1] === "-") {
        return formula[0] - formula[2] === formula[4] ? "O" : "X";
    }
    return formula[0] + formula[2] === formula[4] ? "O" : "X";
  });
}
```
- 풀이 잘봤습니다 quiz를 바로 리턴해줌으로써 불 필요한 result 변수 및 push 메서드를 제거할 수 있습니다.
- 또한 map을 통해 매번 `+`로 형변환 해주는 것을 개선해 좀 더 개독성 좋은 코드로 개선하였습니다.
- 아래는 formula[0], [1], [2] 이렇게 접근하기보다 구조분해 할당을 이용해서 좀 더 명확하게 개선한 풀이입니다. 참고하시길 바랍니다.
```js
function solution(quiz) {
  return quiz.map((el) => {
    const [operand1, operator, operand2, equalSign, formulaResult] = el
      .split(" ")
      .map((el) => (isNaN(+el) ? el : +el));

    const sum =
      operator === "-" ? operand1 + operand2 * -1 : operand1 + operand2;

    return sum === formulaResult ? "O" : "X";
  });
}
```

Reply:
오오오오 훨씬 깔끔한거같아요!!!!  formula[0] + formula[2] === formula[4] ? "O" : "X" 이부분이 좀 신경쓰였는데 감사합니다!!